src/tlv/TlvString.ts

```diff
@@ -42,9 +42,11 @@ export class StringSchema<T extends TlvType.ByteString | TlvType.Utf8String> ext
         return reader.readPrimitive(typeLength) as TlvToPrimitive[T];
     }
 
-    override validate({ length }: TlvToPrimitive[T]): void {
-        if (length > this.maxLength) throw new Error(`String is too long: ${length}, max ${this.maxLength}.`);
-        if (length < this.minLength) throw new Error(`String is too short: ${length}, min ${this.minLength}.`);
+    override validate(value: TlvToPrimitive[T]): void {
+        if (this.type === TlvType.Utf8String && typeof value !== "string") throw new Error(`Expected string, got ${typeof value}.`);
+        if (this.type === TlvType.ByteString && !(value instanceof Uint8Array)) throw new Error(`Expected Uint8Array, got ${typeof value}.`);
```

Comment:
instanceof ByteArray ?

ByteArray is for now an alias for Uint8Array but if this is changed, it should not break this code.